testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/observability/micrometer/MicrometerSetupTask.java

```diff
@@ -15,61 +15,112 @@
  */
 package org.wildfly.test.integration.observability.micrometer;
 
-import static org.junit.Assert.assertEquals;
-
-import java.io.IOException;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.STATISTICS_ENABLED;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
 
 import org.jboss.as.arquillian.api.ServerSetupTask;
 import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
 import org.jboss.as.controller.client.helpers.Operations;
 import org.jboss.as.test.shared.ServerReload;
 import org.jboss.dmr.ModelNode;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.File;
+import java.io.IOException;
 
 public class MicrometerSetupTask implements ServerSetupTask {
+    static OpenTelemetryCollectorContainer otelCollector;
 
     private final ModelNode micrometerExtension = Operations.createAddress("extension", "org.wildfly.extension.micrometer");
     private final ModelNode micrometerSubsystem = Operations.createAddress("subsystem", "micrometer");
+    private boolean extensionAdded = false;
+    private boolean subsystemAdded = false;
 
     @Override
-    public void setup(final ManagementClient managementClient, final String containerId) throws Exception {
-        if (modificationsNeeded()) {
-            execute(managementClient, Operations.createAddOperation(micrometerExtension), true);
+    public void setup(final ManagementClient managementClient, String containerId) throws Exception {
+        startOpenTelemetryCollector();
```

Comment:
Also, I'm pretty sure a ServerSetupTask runs before any @BeforeClass method is invoked, so you need to check AssumeTestGroupUtil.isDockerAvailable().

Reply:
I have that `isDockerAvailable()` call in a `@BeforeClass` in the test class. Will that work there, or should I test here as well?

Reply:
@bstansberry :

> The FT uses of this setup don't need this collector afaict, so I suggest factoring this out or conditionally enabling it somehow.

I can do that, but:
- If the endpoint is configured with an non-functioning URL, the log will be filled with connected refused errors and stack traces
- Starting the collector in the test will mean another restart of WildFly, as I'll need to get the ports exposed by TC and configure WF accordingly. That's probably not a LOT of time, but it's at least something to consider.

If both of those are OK, I'll make the change posthaste.

Reply:
@jasondlee Re refactoring...

The first point is true, although looking at https://ci.wildfly.org/repository/download/WF_MasterLinuxJdk11/363789:id/testsuite/integration/microprofile/target/wildfly/standalone/log/server.log there's only one connection refused log.

Re the 2nd I wasn't suggesting starting it in the test. I meant something like use a separate ServerSetup class for the collector setup. (A test can have more than one ServerSetup; they start in order of declaration and it's LIFO when they stop.

This aspect is ok as is. If the collector is like some other testcontainers uses and leads to intermittent failures we can revisit disabling it for the FT test.

Re isDockerAvailable()....

First, in the @BeforeClass method you need to use assumeDockerAvailable(), not isDockerAvailable().

Second, yes, you need to call isDockerAvailable() in the ServerSetupTask, as the setup() method is called before any @BeforeClass method on the test. Odd, but true. Arquillian even deploys any @Deployment before it will call the @BeforeClass. I think it's a case where @BeforeClass etc are handled by JUnit and Arq does all its stuff before letting JUnit do it's stuff.

To verify my memory I tossed some logging in your branch and this is what came out:

```
22:53:21,242 SEVERE [org.wildfly.test.integration.observability.micrometer.MicrometerSetupTask] (main) Starting OpenTelemetry Container
22:53:21,511 SEVERE [org.wildfly.test.integration.observability.micrometer.MicrometerSetupTask] (main) OpenTelemetry Container failed to start.
22:53:23,941 SEVERE [org.wildfly.test.integration.observability.micrometer.MicrometerOtelIntegrationTestCase] (main) @BeforeClass docker available? false
```